vividus-bdd-engine/src/main/java/org/vividus/bdd/model/RunningStory.java

```diff
@@ -85,18 +87,28 @@ public void setRunningScenario(RunningScenario runningScenario)
         this.runningScenario = runningScenario;
     }
 
-    public String getRunningStep()
+    public String removeRunningStep()
     {
-        return runningStep;
+        return runningSteps.pollLast();
```

Comment:
why not `pop()`

Reply:
Because pop() throws NoSuchELementException if runningSteps is empty

Reply:
Changed on `pollFirst()`